code/model/modelreplace.cpp

```diff
@@ -407,6 +483,229 @@ void VirtualPOFOperationAddSubmodel::process(polymodel* pm, model_read_deferred_
 	}
 }
 
+VirtualPOFOperationAddTurret::VirtualPOFOperationAddTurret() {
+	required_string("+POF to Add:");
+	stuff_string(appendingPOF, F_FILESPEC);
+
+	required_string("+Source Turret:");
+	stuff_string(baseNameSrc, F_NAME);
+
+	required_string("+Destination Turret:");
+	stuff_string(baseNameDest, F_NAME);
+
+	if (optional_string("+Destination Barrel:")) {
```

Comment:
If +Destination Barrel: is specified, will only the barrel be copied?

Reply:
No.
This is less optional then it seems.
Effectively, if you have a single part turret as the source, this will do nothing.
If you have a multipart turret as the source, both base and barrel need a target submodel specified. If you don't, it will warn and abort later during processing of the replacement step.

Reply:
But specifying a source gun turret is not required?

Reply:
Yes.
Since the model will have stored informations about existing barrels in the source base subsys information.
Of if a specified base has a barrel, the modular pof system knows about it, and it does not need to be specified in addition to it

Reply:
Ok.